lib/icinga/checkable-check.cpp

```diff
@@ -428,8 +428,6 @@ void Checkable::ExecuteCheck()
 	double scheduled_start = GetNextCheck();
 	double before_check = Utility::GetTime();
 
-	UpdateNextCheck();
```

Comment:
This change is problematic, this line ensures that Icinga expects the next check at a specific time. The whole freshness logic is built upon this. Please verify and test that freshness still works with this commit.

Reply:
`UpdateNextCheck()` is still called in `ProcessCheckResult()` for active checks, updating the next check in the check execution seems very premature (And would prevent the rest of this patch from working)